packages/block-library/src/image/edit.native.js

```diff
@@ -39,6 +40,27 @@ export default function ImageEdit( { attributes } ) {
 				resizeMethod="scale"
 				source={ { uri: url } }
 			/>
+			{ CaptionTextInput( setAttributes, caption, isSelected ) }
 		</View>
 	);
 }
+
+const CaptionTextInput = ( setAttributes, caption, isSelected ) => {
```

Comment:
At this point it seems that `CaptionTextInput` just checks for caption presence/length besides rendering the caption elements. WDYT about copying what the web-side does and just check and inline the JSX snippet directly, [like this](https://github.com/WordPress/gutenberg/blob/master/packages/block-library/src/image/edit.js#L488-L498)? This will bring us closer to the web-side and will be easier to compare what's different on the mobile.

Reply:
I like that is more declarative in this way, but I agree on trying to be closer to the web-side! I'll push a commit with the change 👍